lib/services/role.go

```diff
@@ -587,6 +613,39 @@ func (set RuleSet) Match(whereParser predicate.Parser, actionsParser predicate.P
 	return false, nil
 }
 
+// matchesWhere returns true if Where rule matches
+// Empty Where block always matches
+func matchesWhere(r *Rule, parser predicate.Parser) (bool, error) {
+	if r.Where == "" {
+		return true, nil
+	}
+	ifn, err := parser.Parse(r.Where)
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	fn, ok := ifn.(predicate.BoolPredicate)
+	if !ok {
+		return false, trace.BadParameter("unsupported type: %T", ifn)
```

Comment:
Is this a user-facing error? I would include the expression and maybe state that it's not a valid predicate instead of showing the compiler type of the function.